src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -72,35 +73,16 @@ public sealed class RequiresUnreferencedCodeAnalyzer : RequiresAnalyzerBase
 		protected override bool IsAnalyzerEnabled (AnalyzerOptions options, Compilation compilation) =>
 			options.IsMSBuildPropertyValueTrue (MSBuildPropertyOptionNames.EnableTrimAnalyzer, compilation);
 
-		protected override ImmutableArray<ISymbol> GetSpecialIncompatibleMembers (Compilation compilation)
-		{
-			var incompatibleMembers = ImmutableArray.CreateBuilder<ISymbol> ();
-			var typeType = compilation.GetTypeByMetadataName ("System.Type");
-			if (typeType != null) {
-				incompatibleMembers.AddRange (typeType.GetMembers ("MakeGenericType").OfType<IMethodSymbol> ());
-			}
-
-			var methodInfoType = compilation.GetTypeByMetadataName ("System.Reflection.MethodInfo");
-			if (methodInfoType != null) {
-				incompatibleMembers.AddRange (methodInfoType.GetMembers ("MakeGenericMethod").OfType<IMethodSymbol> ());
-			}
-
-			var expressionCallType = compilation.GetTypeByMetadataName ("System.Linq.Expressions.Expression");
-			if (expressionCallType != null) {
-				incompatibleMembers.AddRange (expressionCallType.GetMembers ("Call").OfType<IMethodSymbol> ()
-					.Where (m => m.Parameters.Length == 4 && m.Parameters[0] is IParameterSymbol parameterSymbol
-					&& parameterSymbol.Type?.ContainingNamespace?.Name == "System" && parameterSymbol.Type?.Name == "Type"));
-			}
-
-			return incompatibleMembers.ToImmutable ();
-		}
+		/// <summary>
+		/// Add methods that the Trimmer handles intrinsically to an exlusionary list
+		/// </summary>
+		/// <param name="compilation"></param>
+		/// <returns></returns>
 
 		protected override bool ReportSpecialIncompatibleMembersDiagnostic (OperationAnalysisContext operationContext, ImmutableArray<ISymbol> specialIncompatibleMembers, ISymbol member)
 		{
-			if (member is IMethodSymbol && ImmutableArrayOperations.Contains (specialIncompatibleMembers, member, SymbolEqualityComparer.Default)) {
-				// These RUC-annotated APIs are intrinsically handled by the trimmer, which will not produce any
-				// RUC warning related to them. For unrecognized reflection patterns related to generic type/method
-				// creation IL2055/IL2060 should be used instead.
+			// Some RUC-annotated APIs are intrinsically handled by the trimmer
+			if (member is IMethodSymbol method && Intrinsics.GetIntrinsicIdForMethod (new MethodProxy (method)) > IntrinsicId.RequiresReflectionBodyScanner_Sentinel) {
```

Comment:
Per our offline discussion with @tlakollo, instead of the sentinel I think we should check that it's not `IntrinsicId.None`. It probably makes no difference since intrinsics < the sentinel shouldn't have RUC (as far as I understand) but this seems more consistent and avoids introducing terminology from a linker-specific optimization into the analyzer code.